modules/Emails/include/ListView/ListViewDataEmails.php

```diff
@@ -269,209 +191,817 @@ public function getListViewData(
-                                'D, d M Y H:i:s O',
-                                $date
-                            );
-                            if($dateTime  == false) {
-                                // TODO: TASK: UNDEFINED - This needs to be more generic to dealing with different formats from IMAP
-                                $dateTime = DateTime::createFromFormat(
-                                    'd M Y H:i:s O',
-                                    $date
-                                );
+                    SugarVCR::erase($seed->module_dir);
+                    $this->seed =& $seed;
+                    $totalCounted = empty($GLOBALS['sugar_config']['disable_count_query']);
+                    $_SESSION['MAILMERGE_MODULE_FROM_LISTVIEW'] = $seed->module_dir;
+                    if(empty($_REQUEST['action']) || $_REQUEST['action'] != 'Popup'){
+                        $_SESSION['MAILMERGE_MODULE'] = $seed->module_dir;
+                    }
+
+                    $this->setVariableName($seed->object_name, $where, $this->listviewName, $id);
+
+                    $this->seed->id = '[SELECT_ID_LIST]';
+
+                    // if $params tell us to override all ordering
+                    if(!empty($params['overrideOrder']) && !empty($params['orderBy'])) {
+                        $order = $this->getOrderBy(strtolower($params['orderBy']), (empty($params['sortOrder']) ? '' : $params['sortOrder'])); // retreive from $_REQUEST
+                    }
+                    else {
+                        $order = $this->getOrderBy(); // retreive from $_REQUEST
+                    }
+
+                    // still empty? try to use settings passed in $param
+                    if(empty($order['orderBy']) && !empty($params['orderBy'])) {
+                        $order['orderBy'] = $params['orderBy'];
+                        $order['sortOrder'] =  (empty($params['sortOrder']) ? '' : $params['sortOrder']);
+                    }
+
+                    //rrs - bug: 21788. Do not use Order by stmts with fields that are not in the query.
+                    // Bug 22740 - Tweak this check to strip off the table name off the order by parameter.
+                    // Samir Gandhi : Do not remove the report_cache.date_modified condition as the report list view is broken
+                    $orderby = $order['orderBy'];
+                    if (strpos($order['orderBy'],'.') && ($order['orderBy'] != "report_cache.date_modified")) {
+                        $orderby = substr($order['orderBy'],strpos($order['orderBy'],'.')+1);
+                    }
+                    if ($orderby != 'date_entered' && !in_array($orderby, array_keys($filter_fields))) {
+                        $order['orderBy'] = '';
+                        $order['sortOrder'] = '';
+                    }
+
+                    if (empty($order['orderBy'])) {
+                        $orderBy = '';
```

Comment:
See line 374...which sets the order by. However the filters doesn't aways include the order but in the $_REQUEST